typer/main.py

```diff
@@ -165,7 +165,40 @@ def decorator(f: CommandFunctionType) -> CommandFunctionType:
             return f
 
         return decorator
+    
 
+    def async_command(self, *args: Any, backend=None, **kwargs: Any) -> Callable[[CommandFunctionType], CommandFunctionType]:
+        """Same arguments as command but works with async functions."""
+        # Dynamically import either anyio or asyncio
+        if backend is not None:
+            from anyio import run
+        else:
+            from asyncio import run
```

Comment:
Why not use a `try...except ImportError` instead? That way you don't need the `backend` arg.

Reply:
Well I thought about that but what if you need to use the backend arg? Is the backend arg automatic?

Reply:
What do you think actually about this solution:

Instead of adding anyio or any other tool to extras, what if we just had `run_func` as a `kwarg` and default it to `asyncio.run` (as that is builtin). So that anyone can provide literally any tool with any arguments they want.

Reply:
iirc, `asyncio.run` has a different signature to `anyio.run` and `trio.run`, e.g.:
```python
async def foo(bar): ...
asyncio.run(foo(bar))
anyio.run(foo, bar)
trio.run(foo, bar)
```

Reply:
asyncio: `run_func = lambda f, *a, **k: asyncio.run(f(*a, **k))`
anyio: `run_func = lambda f, *a, **k: anyio.run(f, *a, **k)`
trio: `run_func = lambda f, *a, **k: trio.run(f, *a, **k)`

And we just make our run_func take `lambda f, *a, **k`

Reply:
Implemented a demo pending review and feedback.